modules/git/repo_language_stats.go

```diff
@@ -69,8 +89,11 @@ func (repo *Repository) GetLanguageStats(commitID string) (map[string]int64, err
 		return nil, err
 	}
 
-	if len(sizes) == 0 {
-		sizes["other"] = 0
+	// filter special languages unless they are the only language
+	if len(sizes) > 1 {
+		for _, language := range specialLanguages {
+			delete(sizes, language)
```

Comment:
This sounds kind of naive. What happens if there are only markdown and text documents? They will both get deleted.

Reply:
Yes, that's the point.